lib/src/main/scala/spinal/lib/Fragment.scala

```diff
@@ -56,6 +56,36 @@ class StreamFragmentPimped[T <: Data](pimped: Stream[Fragment[T]]) {
 
   def toStreamOfFragment : Stream[T] = pimped.translateWith(pimped.fragment)
 
+/**
+ * Stepwise reduction of all fragments into a single value. With this, you can calculate
+ * the (check)sum of every n elements or parity bits.
+ * @param accumulator A combinatorial function that accumulates the current (param 3) and the previous
+ * 	(param 2) value into the next one (param 1)
+ * @example {{{ outStream = inStream.reduce[SInt]((a, b, c) => { a := b + c }) }}}
```

Comment:
Why not using something similar to the scala collection library ? 

(b, c) => { b + c }   instead of (a, b, c) => { a := b + c } ?

Reply:
I just copied the pattern from `Stream::translateInto` (among others). If we change it, we should change it everywhere to remain consistent. Also, I don't really know how to do it the other way.